app/src/main/java/com/example/geoquiz/QuizActivity.kt

```diff
@@ -0,0 +1,196 @@
+        }
+
+
+
+        questionTextView.setOnClickListener {
+            quizViewModel.moveToNext()
+            updateQuestion()
+        }
+
+        trueButton.setOnClickListener {
+
+            checkAnswer(true)
+
+        }
+
+        falseButton.setOnClickListener {
+
+            checkAnswer(false)
+
+        }
+
+
+        nextButton.setOnClickListener {
+            quizViewModel.moveToNext()
+
+            updateQuestion()
+
+
+        }
+
+        prevButton.setOnClickListener {
+            quizViewModel.moveToPrev()
+            updateQuestion()
+
+        }
+
+        updateQuestion()
+
+        scoreButton.setOnClickListener {
+
+            for (i in correctAnswers) {
+                score += 1
+            }
+
+            val scoreFinalValue=(score.toDouble()/quizViewModel.questionBank.size)*100
+
+
+            Toast.makeText(
+                this,
+                "Your score is: ${scoreFinalValue}%",
+                Toast.LENGTH_SHORT
+            ).show()
+
+        }
+
+        cheatButton.setOnClickListener {
+            val answerIsTrue = quizViewModel.currentQuestionAnswer
+            val intent = CheatActivity.newIntent(this@QuizActivity, answerIsTrue)
+            startActivityForResult(intent, REQUEST_CODE_CHEAT)
+        }
+
+
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode != Activity.RESULT_OK) {
+            return
+        }
+        if (requestCode == REQUEST_CODE_CHEAT) {
+            quizViewModel.isCheater = data?.getBooleanExtra(EXTRA_ANSWER_SHOWN, false) ?: false
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        Log.d(TAG, "onStart() called")
+    }
+
+    override fun onResume() {
+        super.onResume()
+        Log.d(TAG, "onResume() called")
+
+    }
+
+    override fun onPause() {
+        super.onPause()
+        Log.d(TAG, "onPause() called")
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        Log.d(TAG, "onSaveInstanceState() called")
+        outState.putInt(QUESTION_INDEX_KEY, quizViewModel.currentIndex)
+        outState.putBooleanArray(QUESTIONS_ANSWERED_KEY, questionWasAnswered)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        Log.d(TAG, "onStop() called")
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        Log.d(TAG, "onDestroy() called")
+    }
+
+
+    private fun updateQuestion() {
+        val questionTextResId = quizViewModel.currentQuestionText
+        questionTextView.setText(questionTextResId)
+        falseButton.isClickable = !(questionWasAnswered[quizViewModel.currentIndex]!!)
+        trueButton.isClickable = !(questionWasAnswered[quizViewModel.currentIndex]!!)
+
+    }
+
+    private fun checkAnswer(userAnswer: Boolean) {
+        correctAnswers = ArrayList()
```

Comment:
Why is this initialized here? It is initialez and querried, but values are never put into it. Also a good candidate to be moved to ViewModel, I think, depends on it's usage which is not clear currently

Reply:
This correctAnswers is an array that holds the number of questions correct answer, to help me calculate de score.
 I also had a correctAnswers.add(1) in this method each time the userAnswer==answerIsTrue, but it seems that I deleted it somehow. I will fix this.

Reply:
Why do you need a list holding just ones?
Why not count using a simple variable integer? 
@alexandragnj

Reply:
it's correctAnswers.add(messageResId)***, my bad

Reply:
it was the first ideea that came to my mind when I implemented the score button, I suppose I could use only a variable.